aws/product/awsec2product.go

```diff
@@ -0,0 +1,191 @@
+package product
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/trackit/jsonlog"
+	"github.com/trackit/trackit2/db"
+	"github.com/trackit/trackit2/models"
+)
+
+var (
+	// storeAttribute contains the reference between
+	// a field name and the way to store its value.
+	storeAttribute = map[string]func(string, interface{}, *models.AwsProductEc2, jsonlog.Logger){
+		"instanceType":       storeAttributeGenericString,
+		"currentGeneration":  storeAttributeGenericBool,
+		"vcpu":               storeAttributeGenericInt,
+		"memory":             storeAttributeGenericString,
+		"storage":            storeAttributeGenericString,
+		"networkPerformance": storeAttributeGenericString,
+		"tenancy":            storeAttributeGenericString,
+		"operatingSystem":    storeAttributeGenericString,
+		"ecu":                storeAttributeGenericInt,
+		"location":           storeAttributeRegion,
+	}
+)
+
+const (
+	// urlEC2Pricing is the URL used by downloadJSON to
+	// fetch the EC2 pricing.
+	urlEC2Pricing = "https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/index.json"
+)
+
+// storeAttributeRegion creates the region in the database
+// if it doesn't exist. Then, the function stores the region's
+// ID in the struct which will be imported in the database.
+func storeAttributeRegion(name string, value interface{}, dbAwsProduct *models.AwsProductEc2, logger jsonlog.Logger) {
+	dbAwsRegion, err := models.AwsRegionByPretty(db.Db, value.(string))
+	if err != nil {
+		var newDbAwsRegion models.AwsRegion
+		logger.Error("Error during dbAwsRegion fetching", err)
+		newDbAwsRegion.Pretty = value.(string)
+		newDbAwsRegion.Region = value.(string)
+		if err := newDbAwsRegion.Insert(db.Db); err != nil {
+			logger.Error("Error during dbAwsRegion inserting", err)
+		}
+		dbAwsRegion = &newDbAwsRegion
+	}
+	dbAwsProduct.RegionID = dbAwsRegion.ID
+}
+
+// storeAttributeGenericInt stores an int value in the struct
+// which will be imported in the database.
+func storeAttributeGenericInt(name string, value interface{}, dbAwsProduct *models.AwsProductEc2, logger jsonlog.Logger) {
+	if value, ok := value.(string); ok {
+		if value, ok := strconv.Atoi(value); ok == nil {
+			switch name {
+			case "vcpu":
+				dbAwsProduct.Vcpu = value
+			case "ecu":
+				dbAwsProduct.Ecu = value
+			}
+		}
+	}
+}
+
+// storeAttributeGenericBool stores an bool value in the struct
+// which will be imported in the database.
+func storeAttributeGenericBool(name string, value interface{}, dbAwsProduct *models.AwsProductEc2, logger jsonlog.Logger) {
```

Comment:
Your function does not return anything.
Please try to avoid this model as it makes for unmaintainable code quickly.